03b_model/src/run_model.py

```diff
@@ -156,6 +156,13 @@ def select_inputs_targets(inputs, target, train_start_date, test_end_date, out_d
         
         inputs_df = inputs_df.dropna()
     
+    #include lagged inputs, fill NAs with median values
+    if config['include_lagged_input']:
+        inputs_df['discharge_01463500'+'_lag'+str(config['lag'][0])] = inputs_df['discharge_01463500'].shift(config['lag'][0]).fillna(10107.86)
+        inputs_df['discharge_01474500'+'_lag'+str(config['lag'][1])] = inputs_df['discharge_01474500'].shift(config['lag'][1]).fillna(2316.667)
```

Comment:
I read your PR comment as "the lags are hard coded" but maybe you meant that the discharges are hard coded?  Suggestion to change median discharge from hard-coded value to a value that's computed within the script. 

What is the cause of the NAs? Beginning of the record? Gaps in record?

If there are gaps in the record, are the `shift` applied to the days (7 days always equals 7 days) or to the record as-is (7 days could be > 7 days)?

Reply:
> I read your PR comment as "the lags are hard coded" but maybe you meant that the discharges are hard coded? Suggestion to change median discharge from hard-coded value to a value that's computed within the script.

Yeah I guess I meant that the lags are derived from the maximum lagged correlation and that is calculated outside of the pipeline. I'll change the median discharge to calculated in the script.

> What is the cause of the NAs? Beginning of the record? Gaps in record?
Just the beginning of the record. The rest of the record has already been gap filled and checked in previous munge steps

Reply:
Okay, thanks. I think you should trim off the beginning of the record by max(lag) - 1

Reply:
Good call, done in the last commit